airgun/entities/all_hosts.py

```diff
@@ -750,6 +752,118 @@ def read_host_status_icon(self, host_name):
 
         return {'status': icon_status, 'status_details': status_details}
 
+    def manage_traces(
+        self,
+        host_names=None,
+        select_all_hosts=False,
+        traces_to_select=None,
+    ):
+        """
+        This function allows to manage traces for selected hosts through the All Hosts page.
+
+        :param host_names: str with one host or list of hosts to select
+        :param select_all_hosts: bool, if True, all hosts will be selected
+        :param traces_to_select: str with one trace or list of traces to select
+        """
+        if traces_to_select is None:
+            raise ValueError('traces_to_select argument is None! Please provide traces to select!')
+
+        view = self.all_hosts_navigate_and_select_hosts_helper(host_names, select_all_hosts)
+
+        view.bulk_actions_kebab.click()
+        view.bulk_actions_menu.item_select('Manage traces')
+
+        view = ManageTracesModal(self.browser)
+
+        if not isinstance(traces_to_select, list):
+            traces_to_select = [traces_to_select]
+        for trace in traces_to_select:
+            view.search_input.fill(f'application="{trace}"')
+            wait_for(lambda: view.table.is_displayed, timeout=10)
+            row_count = view.table.row_count
+            if row_count == 1:
+                view.table[0][0].widget.fill(True)
+            else:
+                # Select page
+                view.searchbar_dropdown.item_select(view.searchbar_dropdown.items[-2])
+
+        view.restart_btn.click()
+
+        # Instantiate All Hosts view to access the toast alert (without navigation)
+        all_hosts_view = AllHostsTableView(self.browser)
+
+        # Wait for and read the toast alert message
+        self.browser.wait_for_element(all_hosts_view.alert_message, exception=False, timeout=5)
+        if all_hosts_view.alert_message.is_displayed:
+            alert_message_content = all_hosts_view.alert_message.read()
+            all_hosts_view.flash.dismiss()
+            return alert_message_content
+        return None
+
+    def manage_system_purpose(
+        self,
+        host_names=None,
+        select_all_hosts=False,
+        role=None,
+        usage=None,
+        service_level=None,
+        release_version=None,
+    ):
+        """
+        This function allows changing system purpose attributes for selected hosts
+        through the All Hosts page.
+
+        :param host_names: str with one host or list of hosts to select
```

Comment:
Im not requesting a change here but do we usually define each param in our doc strings? looks like AI to me...